src/Conversion/ONNXToKrnl/Sequence/SequenceErase.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------SequenceErase.cpp - Lowering SequenceErase Op-----------------=== //
+//
+// Copyright 2020-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX SequenceErase Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+
+using namespace mlir;
+
+struct ONNXSequenceEraseOpLowering : public ConversionPattern {
+  ONNXSequenceEraseOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(typeConverter,
+            mlir::ONNXSequenceEraseOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op->getLoc();
+    ONNXSequenceEraseOpAdaptor operandAdaptor(operands);
+    ONNXSequenceInsertOp thisOp = dyn_cast<ONNXSequenceInsertOp>(op);
+    MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
+    IndexExprScope IEScope(&rewriter, loc);
+
+    auto input_sequence = operandAdaptor.input_sequence();
+    auto dimSize = rewriter.create<memref::DimOp>(
+        loc, input_sequence, create.math.constantIndex(0));
+    SymbolIndexExpr boundIE(dimSize);
+    auto seqElementType =
+        input_sequence.getType().cast<MemRefType>().getElementType();
+
+    SmallVector<int64_t, 1> dims;
+    // Number of element in seq may be statically known from shape inference
+    dims.emplace_back(thisOp.getResult().getType().cast<SeqType>().getLength());
+    llvm::ArrayRef<int64_t> shape(dims.data(), dims.size());
+    MemRefType outputMemRefType = MemRefType::get(shape, seqElementType);
+
+    auto outputBound = boundIE - 1;
+    SmallVector<IndexExpr, 1> ubsIE;
+    ubsIE.emplace_back(outputBound);
+    Value alloc =
+        insertAllocAndDeallocSimple(rewriter, op, outputMemRefType, loc, ubsIE);
+
+    // Fill the output sequence
+
+    IndexExpr positionIE;
+    if (isFromNone(operandAdaptor.position())) {
+      // Insert at the end of the sequence
+      // Could be optimized as: Copy the input sequence and attach input tensor
+      // at the end But the size for KrnlMemcpy is integer, not Value
+      // memref::copy requires that source and destination have the same shape
+      positionIE = boundIE - 1;
+    } else {
+      positionIE = SymbolIndexExpr(operandAdaptor.position());
+      // Handle the negative position
+      positionIE =
+          IndexExpr::select(positionIE < 0, positionIE + boundIE, positionIE);
+    }
+
+    // Copy before the insert
+    KrnlBuilder createKrnl(rewriter, loc);
```

Comment:
Non necessary because there is already a `KrnlBuilder` declared at line 31

Reply:
I think Ettore means to say that you could add KrnlBuilder to the multi dialect builder.